src/components/CategoryPicker.tsx

```diff
@@ -24,11 +26,18 @@ type CategoryPickerProps = CategoryPickerOnyxProps & {
     policyID: string;
     selectedCategory?: string;
     onSubmit: (item: ListItem) => void;
+
+    /** Whether SectionList should have overflow: "auto" enabled */
+    shouldAddOverflow?: boolean;
 };
 
-function CategoryPicker({selectedCategory, policyCategories, policyRecentlyUsedCategories, policyCategoriesDraft, onSubmit}: CategoryPickerProps) {
+function CategoryPicker({selectedCategory, policyCategories, policyRecentlyUsedCategories, policyCategoriesDraft, onSubmit, shouldAddOverflow = false}: CategoryPickerProps) {
+    const styles = useThemeStyles();
+    const isWeb = getPlatform() === CONST.PLATFORM.WEB;
     const {translate} = useLocalize();
     const [searchValue, debouncedSearchValue, setSearchValue] = useDebouncedState('');
+    // Ensure scrolling works for the SectionList in a nested lists structure involving a Modal on the web.
```

Comment:
Applying specific styles for the SectionList on the web to ensure overflow scrolling works correctly.
This is necessary due to the structure of the component hierarchy:

WorkSpaceCategoriesSettingsPage -> SectionList -> CategoryPicker -> Modal -> SectionList (this is where the styles are applied)

In this setup, the second SectionList inside the Modal does not have scrolling enabled by default.
This is likely caused by the nested SectionList setup, which should not normally be detected as nested due to the Modal, however, it still affects the scrolling behavior.

To resolve this, we conditionally apply overflow and flex styles when rendering on the web.

We could address this in a separate issue investigating why this is happening, but I don't believe this is of high priority ATM